src/frontend/src/catalog/table_catalog.rs

```diff
@@ -634,7 +632,7 @@ impl From<PbTable> for TableCatalog {
                 .map(TableId::from)
                 .collect_vec(),
             cdc_table_id: tb.cdc_table_id,
-            vnode_count: Some(vnode_count), /* from existing (persisted) tables, vnode_count must be set */
+            vnode_count: VnodeCount::set(vnode_count), /* from existing (persisted) tables, vnode_count must be set */
```

Comment:
We obtained the value from `.value()` then construct a `Set` here, which effectively assert that the `maybe_vnode_count` here is not a placeholder value.